src/core/metstate_mod.F90

```diff
@@ -445,268 +445,268 @@ SUBROUTINE Met_Allocate( State_Grid, State_Met, RC)
-      ! CALL CC_CheckVar( 'State_Met%QRAIN', 0, RC )
+      ! ALLOCATE( MetState%QRAIN( GridState%NX, GridState%NY ), STAT=RC )
+      ! CALL CC_CheckVar( 'MetState%QRAIN', 0, RC )
       ! IF ( RC /= CC_SUCCESS ) RETURN
-      ! State_Met%QRAIN = 0e+0_fp
+      ! MetState%QRAIN = 0e+0_fp
 
       ! ! Snow
-      ! ALLOCATE( State_Met%QSNOW( State_Grid%NX, State_Grid%NY ), STAT=RC )
-      ! CALL CC_CheckVar( 'State_Met%QSNOW', 0, RC )
+      ! ALLOCATE( MetState%QSNOW( GridState%NX, GridState%NY ), STAT=RC )
+      ! CALL CC_CheckVar( 'MetState%QSNOW', 0, RC )
       ! IF ( RC /= CC_SUCCESS ) RETURN
-      ! State_Met%QSNOW = 0e+0_fp
+      ! MetState%QSNOW = 0e+0_fp
 
       ! Specific Humidity 2M
-      ALLOCATE( State_Met%QV2M( State_Grid%NX, State_Grid%NY ), STAT=RC )
-      CALL CC_CheckVar( 'State_Met%QV2M', 0, RC )
+      ALLOCATE( MetState%QV2M( GridState%NX, GridState%NY ), STAT=RC )
+      CALL CC_CheckVar( 'MetState%QV2M', 0, RC )
       IF ( RC /= CC_SUCCESS ) RETURN
-      State_Met%QV2M = 0e+0_fp
+      MetState%QV2M = 0e+0_fp
 
       ! Temperature 2M
-      ALLOCATE( State_Met%T2M( State_Grid%NX, State_Grid%NY ), STAT=RC )
-      CALL CC_CheckVar( 'State_Met%T2M', 0, RC )
+      ALLOCATE( MetState%T2M( GridState%NX, GridState%NY ), STAT=RC )
+      CALL CC_CheckVar( 'MetState%T2M', 0, RC )
       IF ( RC /= CC_SUCCESS ) RETURN
-      State_Met%T2M = 0e+0_fp
+      MetState%T2M = 0e+0_fp
 
       ! Skin Temperature
-      ALLOCATE( State_Met%TSKIN( State_Grid%NX, State_Grid%NY ), STAT=RC )
-      CALL CC_CheckVar( 'State_Met%TSKIN', 0, RC )
+      ALLOCATE( MetState%TSKIN( GridState%NX, GridState%NY ), STAT=RC )
+      CALL CC_CheckVar( 'MetState%TSKIN', 0, RC )
       IF ( RC /= CC_SUCCESS ) RETURN
-      State_Met%TSKIN = 0e+0_fp
+      MetState%TSKIN = 0e+0_fp
 
       ! U Wind Speed 10M
-      ALLOCATE( State_Met%U10M( State_Grid%NX, State_Grid%NY ), STAT=RC )
-      CALL CC_CheckVar( 'State_Met%U10M', 0, RC )
+      ALLOCATE( MetState%U10M( GridState%NX, GridState%NY ), STAT=RC )
+      CALL CC_CheckVar( 'MetState%U10M', 0, RC )
       IF ( RC /= CC_SUCCESS ) RETURN
-      State_Met%U10M = 0e+0_fp
+      MetState%U10M = 0e+0_fp
 
       ! V Wind Speed 10M
-      ALLOCATE( State_Met%V10M( State_Grid%NX, State_Grid%NY ), STAT=RC )
-      CALL CC_CheckVar( 'State_Met%V10M', 0, RC )
+      ALLOCATE( MetState%V10M( GridState%NX, GridState%NY ), STAT=RC )
+      CALL CC_CheckVar( 'MetState%V10M', 0, RC )
       IF ( RC /= CC_SUCCESS ) RETURN
-      State_Met%V10M = 0e+0_fp
+      MetState%V10M = 0e+0_fp
 
       ! Surface Roughness
-      ALLOCATE( State_Met%z0( State_Grid%NX, State_Grid%NY ), STAT=RC )
-      CALL CC_CheckVar( 'State_Met%z0', 0, RC )
+      ALLOCATE( MetState%z0( GridState%NX, GridState%NY ), STAT=RC )
+      CALL CC_CheckVar( 'MetState%z0', 0, RC )
       IF ( RC /= CC_SUCCESS ) RETURN
-      State_Met%z0 = 0e+0_fp
+      MetState%z0 = 0e+0_fp
 
       ! Surface Roughness
```

Comment:
If the MetState variables were in a container instead of top-level members, we could do this in a loop instead of individual blocks.